From 771eb27fd71ae08c770944503e76fe54605e3372 Mon Sep 17 00:00:00 2001
From: Florinel Iordache <florinel.iordache@nxp.com>
Date: Mon, 4 Feb 2019 09:45:54 +0200
Subject: [PATCH 051/235] sdk_fman: avoid array overflow error in fman port
 init

Perform a verification of external buffer pools used which can cause array
overflow error in port init function SetExtBufferPools() if it was set to
value 1 via fman port API

Signed-off-by: Florinel Iordache <florinel.iordache@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../ethernet/freescale/sdk_fman/Peripherals/FM/Port/fm_port.c  | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Port/fm_port.c b/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Port/fm_port.c
index ec6e0ed59376..6df80b31d63c 100644
--- a/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Port/fm_port.c
+++ b/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Port/fm_port.c
@@ -581,7 +581,8 @@ static t_Error SetExtBufferPools(t_FmPort *p_FmPort)
     p_FmPort->rxPoolsParams.numOfPools = p_ExtBufPools->numOfPoolsUsed;
     p_FmPort->rxPoolsParams.largestBufSize =
             sizesArray[orderedArray[p_ExtBufPools->numOfPoolsUsed - 1]];
-    p_FmPort->rxPoolsParams.secondLargestBufSize =
+    if (p_ExtBufPools->numOfPoolsUsed > 1)
+        p_FmPort->rxPoolsParams.secondLargestBufSize =
             sizesArray[orderedArray[p_ExtBufPools->numOfPoolsUsed - 2]];
 
     /* FMBM_RMPD reg. - pool depletion */
-- 
2.17.1

